aea/connections/base.py

```diff
@@ -40,17 +44,30 @@ def __init__(self):
 class Connection(ABC):
     """Abstract definition of a connection."""
 
-    def __init__(self, connection_id: str):
+    def __init__(self, connection_id: Optional[str] = None, supported_protocols: Optional[Set[str]] = None):
         """
         Initialize the connection.
 
         :param connection_id: the connection identifier.
+        :param supported_protocols: the set of protocol ids supported by the connection.
         """
-        self._connection_id = connection_id
+        self._connection_id = connection_id if connection_id is not None else type(self).__name__
+        self._supported_protocols = self._get_supported_protocols(supported_protocols)
 
         self._loop = None  # type: Optional[AbstractEventLoop]
         self._connection_status = ConnectionStatus()
 
+        if len(self.supported_protocols) == 0:
+            logger.warning("No supported protocol for connection '{}'".format(connection_id))
+
+    def _get_supported_protocols(self, supported_protocols: Optional[Set[str]] = None) -> Set[str]:
```

Comment:
It works as follows:
- if the user specifies `supported_protocols` in the constructor, set the supported protocols to that value
- else, if the connection class has an attribute `supported_protocols` (some examples below) and it is a `set`, then use it
- else, return the empty set. A warning is logged to notify the user of this corner case.

Reply:
See the comments below re `restricted_to_protocols`